test/controllers/pieces_controller_test.rb

```diff
@@ -1,4 +1,12 @@
 require 'test_helper'
 
 class PiecesControllerTest < ActionController::TestCase
+  test "should get update" do
+    piece = FactoryGirl.create(:piece)
+    put :update, :id => piece.id, :piece => { :x_position => 2, :y_position => 1 }
```

Comment:
right after this, add `piece.reload`

**TLDR** ActiveRecord caches attributes on the ruby model object so that it doesn't have to keep asking the server for details. When we create send the request to update the record, the model ( ruby object ) and record (postgresql row) are out of sync. `reload` method updates the data in the model ( ruby object ).